lib/mistral_ai_summary_example/summary_settings_page.dart

```diff
@@ -1,183 +1,287 @@
+                      .read<SummarySettingsModel>()
+                      .setTemperature(value),
+                ),
+              ],
             ),
-            ButtonBar(
+            Column(
+              crossAxisAlignment: CrossAxisAlignment.start,
               children: [
-                ElevatedButton(
-                  onPressed: () {
-                    Navigator.pop(context, summarySettings);
-                  },
-                  child: const Text('Save'),
+                Text(
+                  'Top P: ${topP.toStringAsFixed(2)}',
+                ),
+                CustomSlider(
+                  value: topP,
+                  label: topP.toStringAsFixed(2),
+                  onChanged: (value) =>
+                      context.read<SummarySettingsModel>().setTopP(value),
                 ),
               ],
             ),
+            const Text(
+              'We recommend altering only one, not both',
+              style: TextStyle(
+                fontSize: 12,
+                color: Colors.grey,
+              ),
+            ),
           ],
         ),
       ),
     );
   }
 }
+
+class MaxTokensSettingWidget extends StatelessWidget {
+  const MaxTokensSettingWidget({
+    super.key,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return ClickableSettingsItem(
+      settingName: 'Max tokens',
+      settingValue: context
+              .watch<SummarySettingsModel>()
+              .settings
+              .maxTokens
+              ?.toString() ??
+          'Unlimited',
+      onTap: () {
+        final summarySettingsModel = context.read<SummarySettingsModel>();
+        showDialog<void>(
+          context: context,
+          builder: (_) => ChangeNotifierProvider.value(
+            value: summarySettingsModel,
+            child: const MaxTokensDialog(),
+          ),
+        );
+      },
+    );
+  }
+}
+
+class RandomSeedSettingWidget extends StatelessWidget {
+  const RandomSeedSettingWidget({
+    super.key,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return ClickableSettingsItem(
+      settingName: 'Random seed',
+      settingValue: context
+              .watch<SummarySettingsModel>()
+              .settings
+              .randomSeed
+              ?.toString() ??
+          'Unset',
+      onTap: () {
+        final summarySettingsModel = context.read<SummarySettingsModel>();
+        showDialog<void>(
+          context: context,
+          builder: (_) => ChangeNotifierProvider.value(
+            value: summarySettingsModel,
+            child: const RandomSeedDialog(),
+          ),
+        );
+      },
+    );
+  }
+}
+
+class SafePromptSettingWidget extends StatelessWidget {
+  const SafePromptSettingWidget({
+    super.key,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    final settingValue =
+        context.watch<SummarySettingsModel>().settings.safePrompt;
+    return ColoredBox(
```

Comment:
I think you could use just a `SwitchListTile` here.

Reply:
You're right, refactored in [bf91385](https://github.com/nomtek/flutter_ai_examples/pull/30/commits/bf91385b972a64f8afffa2e1232f5a3574f806b0)